Applications/Spire/Source/OrderImbalanceIndicator/CachedOrderImbalanceIndicatorModel.cpp

```diff
@@ -17,20 +17,27 @@ CachedOrderImbalanceIndicatorModel::CachedOrderImbalanceIndicatorModel(
   m_subscription_promise = std::move(promise);
 }
 
-QtPromise<std::vector<Nexus::OrderImbalance>>
-    CachedOrderImbalanceIndicatorModel::load(
-    const TimeInterval& interval) {
+QtPromise<std::vector<OrderImbalance>>
+    CachedOrderImbalanceIndicatorModel::load(const TimeInterval& interval) {
   if(boost::icl::contains(m_intervals, interval)) {
     return load_from_cache(interval);
   }
-  return load_from_model(interval);
+  return load_from_model(interval).then([=] (const auto& loaded_imbalances) {
+    on_imbalances_loaded(interval, loaded_imbalances.Get());
+    return load_from_cache(interval);
+  });
 }
 
-QtPromise<std::vector<Nexus::OrderImbalance>>
-    CachedOrderImbalanceIndicatorModel::load(
-    const Nexus::Security& security,
+QtPromise<std::vector<OrderImbalance>>
+    CachedOrderImbalanceIndicatorModel::load(const Nexus::Security& security,
     const TimeInterval& interval) {
-  return QtPromise([] { return std::vector<OrderImbalance>(); });
+  if(boost::icl::contains(m_intervals, interval)) {
+    return m_cache.load(security, interval);
+  }
+  return load_from_model(interval).then([=] (const auto& loaded_imbalances) {
```

Comment:
You must only load the imbalances for ```security``` instead of loading imbalances for everything.

Reply:
I thought a 'greedy' approach to caching would be beneficial because it's caching data that could be used by every subscriber in the future.

Should I just not cache the data and call the source model's load method directly?

The alternative is to restructure the cache so it can cache different securities separately, but I'm not sure if it's worth the effort it would take.